autocertkit/utils.py

```diff
@@ -945,6 +946,22 @@ def is_vf_disabled(session):
     return sum == 0
 
 
+def get_vf_driver_info(session, host, vm_ref, device):
+    cmd = b"%s -i %s" % (ETHTOOL, device)
+    cmd = binascii.hexlify(cmd)
+    res = call_ack_plugin(session, 'shell_run',
+                          {'cmd': cmd, 'vm_ref': vm_ref,
+                           'username': 'root', 'password': DEFAULT_PASSWORD},
+                          host)
+    ret = {}
+    for line in res.pop()['stdout'].split('\n'):
+        if line.startswith(('driver:', 'version:', 'bus-info:')):
```

Comment:
Can we not use a regex instead? would be easier to see what you're doing and feel much safer.

Reply:
Below is sample of output, startswith() is best suited to check string with handle. I learnt function in xenrt code, it's used widely there.
re is powerful indeed, while here needless to concern the value contains
```
driver: e1000e
version: 3.3.5-NAPI
firmware-version: 1.8-0
bus-info: 0000:08:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: no
```

Reply:
My opinion would be that a regex makes what you are intending to do clearer. If you want to keep it, perhaps you make a helper function which turns a list of `key: value` into a dictionary?

Reply:
Understood now! I believe you are very good at RE, and not all people like to read those metacharacter in RE. I will fix.